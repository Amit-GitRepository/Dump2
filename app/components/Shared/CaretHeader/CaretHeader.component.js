import BottomCaretView from '../BottomCaretView/BottomCaretView.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CaretHeader.styles.js';
import Touchable from '../Touchable/Touchable.component';
import {colors} from '../../../themes/constants.styles';
import {ISText} from '../../Shared';
import {View} from 'react-native';
import * as Animatable from 'react-native-animatable';

const AnimatableBottomCaretView = Animatable.createAnimatableComponent(BottomCaretView);

// CaretHeader: To be used in most of the cards with header.
// View, children and text style can be overridden by props.

class CaretHeader extends Component {
  render () {
    const {active, onPress, text, wrapperStyles, iconSize, activeStyles, inActiveStyles, titleWrapperStyle, caretColor, children, caretTextType} = this.props;
    const titleContentStyle = active ? [styles.titleContent, activeStyles] : [styles.titleContent, inActiveStyles];
    const WrapperView = onPress ? Touchable : View;
    const ActiveView = active ? AnimatableBottomCaretView : Animatable.View;
    const titleStyle = active ? [styles.title, styles.activeTitle] : styles.title;
    const titleContainerStyles = [styles.titleContainer, {borderBottomColor: caretColor}];

    return (
      <WrapperView style={wrapperStyles} onPress={onPress}>
        <ActiveView useNativeDriver animation='fadeIn' duration={500} caretColor={caretColor} iconSize={iconSize} style={styles.caretShadow}>
          <View style={titleContainerStyles}>
            <View style={[titleContentStyle, titleWrapperStyle]}>
              {text ? <ISText style={titleStyle} type={caretTextType}>{text}</ISText> : null}
              {children}
            </View>
          </View>
        </ActiveView>
      </WrapperView>
    );
  }
}

CaretHeader.defaultProps = {
  active: false,
  onPress: null,
  text: '',
  wrapperStyles: {},
  activeStyles: styles.activeStyle,
  inActiveStyles: {},
  titleWrapperStyle: {},
  iconSize: 15,
  caretColor: colors.PRIMARY_ACTIONABLE,
  caretTextType: 'REGULAR'
};
CaretHeader.propTypes = {
  active: PropTypes.bool,
  onPress: PropTypes.func,
  text: PropTypes.string,
  wrapperStyles: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.array
  ]),
  activeStyles: PropTypes.object,
  inActiveStyles: PropTypes.object,
  titleWrapperStyle: PropTypes.object,
  iconSize: PropTypes.number,
  caretColor: PropTypes.string,
  children: PropTypes.node,
  caretTextType: PropTypes.string
};
export default CaretHeader;
