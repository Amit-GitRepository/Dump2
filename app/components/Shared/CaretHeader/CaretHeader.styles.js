import {colors, fonts} from '../../../themes/constants.styles';
import {shadowActiveStyle} from '../../../themes/application.styles';

export default {
  titleContainer: {
    paddingHorizontal: 5,
    borderBottomColor: colors.PRIMARY_ACTIONABLE
  },
  caretShadow: {
    elevation: 3,
    borderWidth: 0, // Without borderWidth: 0 shadow is not rendering in Android
    ...shadowActiveStyle
  },
  activeStyle: {
    backgroundColor: colors.PRIMARY_ACTIONABLE
  },
  titleContent: {
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    padding: 10,
    alignItems: 'center'
  },
  title: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    color: colors.SECONDARY_BG_TEXT_CONTRAST
  },
  activeTitle: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  }
};
