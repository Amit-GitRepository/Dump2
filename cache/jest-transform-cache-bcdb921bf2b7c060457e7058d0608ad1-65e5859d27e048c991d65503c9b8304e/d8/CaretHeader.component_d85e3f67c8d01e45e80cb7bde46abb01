Object.defineProperty(exports,"__esModule",{value:true});var cov_1hreq5bywn=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CaretHeader/CaretHeader.component.js',hash='87cbad44b483a914754819311905c26e1572ff20',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CaretHeader/CaretHeader.component.js',statementMap:{'0':{start:{line:11,column:34},end:{line:11,column:87}},'1':{start:{line:18,column:158},end:{line:18,column:168}},'2':{start:{line:19,column:30},end:{line:19,column:114}},'3':{start:{line:20,column:24},end:{line:20,column:50}},'4':{start:{line:21,column:23},end:{line:21,column:75}},'5':{start:{line:22,column:23},end:{line:22,column:81}},'6':{start:{line:23,column:33},end:{line:23,column:89}},'7':{start:{line:24,column:4},end:{line:35,column:6}},'8':{start:{line:39,column:0},end:{line:50,column:2}},'9':{start:{line:51,column:0},end:{line:64,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:17,column:2},end:{line:17,column:3}},loc:{start:{line:17,column:12},end:{line:36,column:3}},line:17}},branchMap:{'0':{loc:{start:{line:19,column:30},end:{line:19,column:114}},type:'cond-expr',locations:[{start:{line:19,column:39},end:{line:19,column:74}},{start:{line:19,column:77},end:{line:19,column:114}}],line:19},'1':{loc:{start:{line:20,column:24},end:{line:20,column:50}},type:'cond-expr',locations:[{start:{line:20,column:34},end:{line:20,column:43}},{start:{line:20,column:46},end:{line:20,column:50}}],line:20},'2':{loc:{start:{line:21,column:23},end:{line:21,column:75}},type:'cond-expr',locations:[{start:{line:21,column:32},end:{line:21,column:57}},{start:{line:21,column:60},end:{line:21,column:75}}],line:21},'3':{loc:{start:{line:22,column:23},end:{line:22,column:81}},type:'cond-expr',locations:[{start:{line:22,column:32},end:{line:22,column:66}},{start:{line:22,column:69},end:{line:22,column:81}}],line:22},'4':{loc:{start:{line:29,column:15},end:{line:29,column:92}},type:'cond-expr',locations:[{start:{line:29,column:22},end:{line:29,column:85}},{start:{line:29,column:88},end:{line:29,column:92}}],line:29}},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0},f:{'0':0},b:{'0':[0,0],'1':[0,0],'2':[0,0],'3':[0,0],'4':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _BottomCaretView=require('../BottomCaretView/BottomCaretView.component');var _BottomCaretView2=_interopRequireDefault(_BottomCaretView);
var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _CaretHeaderStyles=require('./CaretHeader.styles.js');var _CaretHeaderStyles2=_interopRequireDefault(_CaretHeaderStyles);
var _Touchable=require('../Touchable/Touchable.component');var _Touchable2=_interopRequireDefault(_Touchable);
var _constants=require('../../../themes/constants.styles');
var _Shared=require('../../Shared');
var _reactNative=require('react-native');
var _reactNativeAnimatable=require('react-native-animatable');var Animatable=_interopRequireWildcard(_reactNativeAnimatable);function _interopRequireWildcard(obj){if(obj&&obj.__esModule){return obj;}else{var newObj={};if(obj!=null){for(var key in obj){if(Object.prototype.hasOwnProperty.call(obj,key))newObj[key]=obj[key];}}newObj.default=obj;return newObj;}}function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}

var AnimatableBottomCaretView=(cov_1hreq5bywn.s[0]++,Animatable.createAnimatableComponent(_BottomCaretView2.default));var




CaretHeader=function(_Component){_inherits(CaretHeader,_Component);function CaretHeader(){_classCallCheck(this,CaretHeader);return _possibleConstructorReturn(this,(CaretHeader.__proto__||Object.getPrototypeOf(CaretHeader)).apply(this,arguments));}_createClass(CaretHeader,[{key:'render',value:function render()
{cov_1hreq5bywn.f[0]++;var _ref=(cov_1hreq5bywn.s[1]++,
this.props),active=_ref.active,onPress=_ref.onPress,text=_ref.text,wrapperStyles=_ref.wrapperStyles,iconSize=_ref.iconSize,activeStyles=_ref.activeStyles,inActiveStyles=_ref.inActiveStyles,titleWrapperStyle=_ref.titleWrapperStyle,caretColor=_ref.caretColor,elevation=_ref.elevation,children=_ref.children,caretTextType=_ref.caretTextType;
var titleContentStyle=(cov_1hreq5bywn.s[2]++,active?(cov_1hreq5bywn.b[0][0]++,[_CaretHeaderStyles2.default.titleContent,activeStyles]):(cov_1hreq5bywn.b[0][1]++,[_CaretHeaderStyles2.default.titleContent,inActiveStyles]));
var WrapperView=(cov_1hreq5bywn.s[3]++,onPress?(cov_1hreq5bywn.b[1][0]++,_Touchable2.default):(cov_1hreq5bywn.b[1][1]++,_reactNative.View));
var ActiveView=(cov_1hreq5bywn.s[4]++,active?(cov_1hreq5bywn.b[2][0]++,AnimatableBottomCaretView):(cov_1hreq5bywn.b[2][1]++,Animatable.View));
var titleStyle=(cov_1hreq5bywn.s[5]++,active?(cov_1hreq5bywn.b[3][0]++,[_CaretHeaderStyles2.default.title,_CaretHeaderStyles2.default.activeTitle]):(cov_1hreq5bywn.b[3][1]++,_CaretHeaderStyles2.default.title));
var titleContainerStyles=(cov_1hreq5bywn.s[6]++,[_CaretHeaderStyles2.default.titleContainer,{borderBottomColor:caretColor}]);cov_1hreq5bywn.s[7]++;
return(
_react2.default.createElement(WrapperView,{style:wrapperStyles,onPress:onPress},
_react2.default.createElement(ActiveView,{useNativeDriver:true,animation:'fadeIn',duration:500,caretColor:caretColor,iconSize:iconSize},
_react2.default.createElement(_reactNative.View,{style:titleContainerStyles,elevation:elevation},
_react2.default.createElement(_reactNative.View,{style:[titleContentStyle,titleWrapperStyle]},
text?(cov_1hreq5bywn.b[4][0]++,_react2.default.createElement(_Shared.ISText,{style:titleStyle,type:caretTextType},text)):(cov_1hreq5bywn.b[4][1]++,null),
children)))));





}}]);return CaretHeader;}(_react.Component);cov_1hreq5bywn.s[8]++;


CaretHeader.defaultProps={
active:false,
onPress:null,
text:'',
wrapperStyles:{},
activeStyles:_CaretHeaderStyles2.default.activeStyle,
inActiveStyles:{},
titleWrapperStyle:{},
iconSize:15,
caretColor:_constants.colors.PRIMARY_ACTIONABLE,
caretTextType:'REGULAR'};cov_1hreq5bywn.s[9]++;

CaretHeader.propTypes={
active:_propTypes2.default.bool,
onPress:_propTypes2.default.func,
text:_propTypes2.default.string,
wrapperStyles:_propTypes2.default.object,
activeStyles:_propTypes2.default.object,
inActiveStyles:_propTypes2.default.object,
titleWrapperStyle:_propTypes2.default.object,
iconSize:_propTypes2.default.number,
elevation:_propTypes2.default.number,
caretColor:_propTypes2.default.string,
children:_propTypes2.default.node,
caretTextType:_propTypes2.default.string};exports.default=

CaretHeader;