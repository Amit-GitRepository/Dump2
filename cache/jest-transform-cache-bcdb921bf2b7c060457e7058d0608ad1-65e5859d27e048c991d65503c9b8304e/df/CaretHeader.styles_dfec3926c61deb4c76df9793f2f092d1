Object.defineProperty(exports,"__esModule",{value:true});var cov_ojy3zdl84=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CaretHeader/CaretHeader.styles.js',hash='4ce7f12ff58dd7d2b3ea85b108de491ae55e54c2',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/CaretHeader/CaretHeader.styles.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _constants=require('../../../themes/constants.styles');exports.default=

{
titleContainer:{
paddingHorizontal:5,
borderBottomWidth:3,
borderBottomColor:_constants.colors.PRIMARY_ACTIONABLE},

activeStyle:{
backgroundColor:_constants.colors.PRIMARY_ACTIONABLE},

titleContent:{
backgroundColor:_constants.colors.PRIMARY_BG_SEPARATOR,
padding:10,
alignItems:'center',
borderTopLeftRadius:5,
borderTopRightRadius:5},

title:{
fontSize:_constants.fonts.FONT_SIZE_MEDIUM,
color:_constants.colors.SECONDARY_BG_TEXT_CONTRAST},

activeTitle:{
color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST}};