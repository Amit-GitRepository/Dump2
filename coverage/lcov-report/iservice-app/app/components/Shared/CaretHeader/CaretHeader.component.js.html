<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Shared/CaretHeader/CaretHeader.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Shared/CaretHeader</a> CaretHeader.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>10/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">90% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>9/10</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>10/10</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">125x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-yes">25x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">125x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">125x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import BottomCaretView from '../BottomCaretView/BottomCaretView.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './CaretHeader.styles.js';
import Touchable from '../Touchable/Touchable.component';
import {colors} from '../../../themes/constants.styles';
import {ISText} from '../../Shared';
import {View} from 'react-native';
import * as Animatable from 'react-native-animatable';
&nbsp;
const AnimatableBottomCaretView = Animatable.createAnimatableComponent(BottomCaretView);
&nbsp;
// CaretHeader: To be used in most of the cards with header.
// View, children and text style can be overridden by props.
&nbsp;
class CaretHeader extends Component {
  render () {
    const {active, onPress, text, wrapperStyles, iconSize, activeStyles, inActiveStyles, titleWrapperStyle, caretColor, elevation, children, caretTextType} = this.props;
    const titleContentStyle = active ? [styles.titleContent, activeStyles] : [styles.titleContent, inActiveStyles];
    const WrapperView = onPress ? <span class="branch-0 cbranch-no" title="branch not covered" >Touchable </span>: View;
    const ActiveView = active ? AnimatableBottomCaretView : Animatable.View;
    const titleStyle = active ? [styles.title, styles.activeTitle] : styles.title;
    const titleContainerStyles = [styles.titleContainer, {borderBottomColor: caretColor}];
    return (
      &lt;WrapperView style={wrapperStyles} onPress={onPress}&gt;
        &lt;ActiveView useNativeDriver animation='fadeIn' duration={500}  caretColor={caretColor} iconSize={iconSize}&gt;
          &lt;View style={titleContainerStyles} elevation={elevation}&gt;
            &lt;View style={[titleContentStyle, titleWrapperStyle]}&gt;
              {text ? &lt;ISText style={titleStyle} type={caretTextType}&gt;{text}&lt;/ISText&gt; : null}
              {children}
            &lt;/View&gt;
          &lt;/View&gt;
        &lt;/ActiveView&gt;
      &lt;/WrapperView&gt;
    );
  }
}
&nbsp;
CaretHeader.defaultProps = {
  active: false,
  onPress: null,
  text: '',
  wrapperStyles: {},
  activeStyles: styles.activeStyle,
  inActiveStyles: {},
  titleWrapperStyle: {},
  iconSize: 15,
  caretColor: colors.PRIMARY_ACTIONABLE,
  caretTextType: 'REGULAR'
};
CaretHeader.propTypes = {
  active: PropTypes.bool,
  onPress: PropTypes.func,
  text: PropTypes.string,
  wrapperStyles: PropTypes.object,
  activeStyles: PropTypes.object,
  inActiveStyles: PropTypes.object,
  titleWrapperStyle: PropTypes.object,
  iconSize: PropTypes.number,
  elevation: PropTypes.number,
  caretColor: PropTypes.string,
  children: PropTypes.node,
  caretTextType: PropTypes.string
};
export default CaretHeader;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
