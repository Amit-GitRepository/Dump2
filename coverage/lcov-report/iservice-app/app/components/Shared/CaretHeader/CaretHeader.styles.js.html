<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Shared/CaretHeader/CaretHeader.styles.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Shared/CaretHeader</a> CaretHeader.styles.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import {colors, fonts} from '../../../themes/constants.styles';
import {shadowActiveStyle} from '../../../themes/application.styles';
&nbsp;
export default {
  titleContainer: {
    paddingHorizontal: 5,
    borderBottomWidth: 3,
    borderBottomColor: colors.PRIMARY_ACTIONABLE,
    ...shadowActiveStyle
  },
  activeStyle: {
    backgroundColor: colors.PRIMARY_ACTIONABLE
  },
  titleContent: {
    backgroundColor: colors.PRIMARY_BG_SEPARATOR,
    padding: 10,
    alignItems: 'center',
    borderTopLeftRadius: 5,
    borderTopRightRadius: 5
  },
  title: {
    fontSize: fonts.FONT_SIZE_MEDIUM,
    color: colors.SECONDARY_BG_TEXT_CONTRAST
  },
  activeTitle: {
    color: colors.PRIMARY_BG_TEXT_CONTRAST
  }
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
