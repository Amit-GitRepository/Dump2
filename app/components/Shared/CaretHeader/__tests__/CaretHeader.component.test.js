import CaretHeader from '../CaretHeader.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('CaretHeader component', () => {
  it('should render correctly', () => {
    const componentTree = renderer.create(<CaretHeader />).toJSON();
    expect(componentTree).toMatchSnapshot();
  });
});
