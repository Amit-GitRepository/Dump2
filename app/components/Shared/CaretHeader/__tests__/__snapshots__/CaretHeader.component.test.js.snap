// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaretHeader component should render correctly 1`] = `
<View
  onPress={null}
  style={Object {}}
>
  <View
    animation="fadeIn"
    caretColor="#F93E60"
    duration={500}
    iconSize={15}
    style={
      Object {
        "borderWidth": 0,
        "elevation": 3,
        "shadowColor": "#303C46",
        "shadowOffset": Object {
          "height": 2,
          "width": 1,
        },
        "shadowOpacity": 0.3,
        "shadowRadius": 1,
      }
    }
    useNativeDriver={true}
  >
    <View
      style={
        Array [
          Object {
            "borderBottomColor": "#F93E60",
            "paddingHorizontal": 5,
          },
          Object {
            "borderBottomColor": "#F93E60",
          },
        ]
      }
    >
      <View
        style={
          Array [
            Array [
              Object {
                "alignItems": "center",
                "backgroundColor": "#E9E9EA",
                "padding": 10,
              },
              Object {},
            ],
            Object {},
          ]
        }
      />
    </View>
  </View>
</View>
`;
